AWSTranscribeStreaming/Internal/AWSTranscribeStreamingWebSocketProvider.h

```diff
@@ -0,0 +1,34 @@
+//
+// Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// A copy of the License is located at
+//
+// http://aws.amazon.com/apache2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+
+#import <Foundation/Foundation.h>
+#import "AWSTranscribeStreamingWebSocketProviderDelegate.h"
+#import "AWSSRWebSocketDelegateAdaptor.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol AWSTranscribeStreamingWebSocketProvider <NSObject>
```

Comment:
Have you looked at these existing protocols? 

@royjit refactored AppSyncClient to use the AppSyncWebsocketProvider protocol: https://github.com/awslabs/aws-mobile-appsync-sdk-ios/tree/master/AWSAppSyncClient/Subscription/Websocket

Then I moved it over to Amplify for APIPlugin's subscriptions: https://github.com/aws-amplify/amplify-ios/tree/master/AmplifyPlugins/API/AWSAPICategoryPlugin/Subscription/Websocket
AppSync

Reply:
the differences,
- `isConnected: Bool` 
- AppSync's `connect` takes in optional delgate, rather than have a `setDelegateAndDelegateDispatchQueue`
- Amplify.API exposes a setListener with closure instead of delegate pattern